src/libNode/Node.cpp

```diff
@@ -625,7 +627,8 @@ bool Node::ProcessTxnPacketFromLookup(
         return false;
     }
 
-    if (m_mediator.m_currentEpochNum % NUM_FINAL_BLOCK_PER_POW == 0
+    if (m_mediator.m_consensusID
```

Comment:
Looks like the logic changed here but I can't understand why. Can you confirm it's what you intended?

Reply:
Yes, this is for realizing `"Once the PoW is done and a new DS block is broadcast, the network should continue with the last valid txn epoch number, thus the way of determining vacuous epoch will change."`. For example, we have 50 txBlocks per DS epoch, if the fallback happens at epoch 30, then the next vacuous epoch will be on epoch 80, this is determined by the consensus id, as we reset it after fallback, so it will increases from 0 at epoch 30 to 50 at epoch 80.

Reply:
What if we kept the current logic and just had fewer tx blocks in this particular epoch? Anyway it is an extraordinary case as fallback shouldn't happen occasionally. My concern is that the link between consensus ID and the epoch stuff is a very loose one. If we randomize the ID in the future then this logic will be broken.

Reply:
Ok I think that make sense, and also doable, will change